public/app/features/logs/components/LogRowMessage.tsx

```diff
@@ -94,16 +104,23 @@ export class LogRowMessage extends PureComponent<Props> {
             className={cx('log-row-menu', styles.rowMenu, {
               [styles.rowMenuWithContextButton]: shouldShowContextToggle,
             })}
-            onClick={(e) => e.stopPropagation()}
+            onClick={this.onLogRowClick}
           >
             {shouldShowContextToggle && (
               <Tooltip placement="top" content={'Show context'}>
                 <IconButton size="md" name="gf-show-context" onClick={this.onShowContextClick} />
               </Tooltip>
             )}
-            <Tooltip placement="top" content={'Copy'}>
-              <IconButton size="md" name="copy" onClick={() => navigator.clipboard.writeText(restructuredEntry)} />
-            </Tooltip>
+            <ClipboardButton
+              className={styles.copyLogButton}
+              icon="copy"
+              variant="secondary"
+              fill="text"
+              size="sm"
+              getText={this.getLogText}
```

Comment:
As we already have `restructuredEntry` we can just do `() => restructuredEntry`, so we don't have to restructure it again.

Reply:
Since it's the log row message component, I did this change as a small performance improvement, to prevent creating new arrow functions which cause unnecessary re-renders. Both `onLogRowClick` and `getLogText` would be created once per log line only once.
In any case, it's a simple change so I don't mind reverting it.

Reply:
👍